pkg/server/backuprestoreserver.go

```diff
@@ -238,7 +240,8 @@ func (b *BackupRestoreServer) runEtcdProbeLoopWithSnapshotter(ctx context.Contex
 		gcStopCh := make(chan struct{})
 		go ssr.RunGarbageCollector(gcStopCh)
 		b.logger.Infof("Starting snapshotter...")
-		if err := ssr.Run(ssrStopCh, initialDeltaSnapshotTaken); err != nil {
+		startWithFullSnapshot := !(initialDeltaSnapshotTaken && time.Since(ssr.PrevFullSnapshot.CreatedOn).Hours() <= recentFullSnapshotPeriodInHours)
```

Comment:
I have a query:
Suppose there was no change in the revision since the last full snapshot, then in this function it takes a full snapshot and proceeds taking delta snapshots. Say, etcd gets OOM killed here. Wont the cycle reappear as we wont take delta snapshots as [here](https://github.com/gardener/etcd-backup-restore/pull/222/files#diff-63ebe6acc9f30c180e7bf89c61edbc97R223) we are not checking for recent full snapshot. If deltaSnaps are not there and if full snapshots are recent then how do we proceed?

Reply:
Nope.
On next restart, it will attempt delta snap first. So, `initialDeltaSnapshotTaken` will be true and last full snap will be within 24hr. So, it won't take full snapshot immediately but as per the full snap schedule.